Model/Customer.php

```diff
@@ -0,0 +1,167 @@
+
+        $this->initializeObject();
+    }
+
+    /**
+     * Check whether the executor is logged in or not.
+     *
+     * @return bool
+     */
+    public function isLoggedIn()
+    {
+        return $this->getMagentoCustomerId() ? true : false;
+    }
+
+    /**
+     * Get an Omise customer id that belongs to
+     * a specific user who execute the method.
+     *
+     * @return string
+     */
+    public function getId()
+    {
+        return $this->magentoCustomer->getData(self::OMISE_CUSTOMER_ID_FIELD);
+    }
+
+    /**
+     * Get a Magento customer id that belongs to
+     * a specific user who execute the method.
+     *
+     * @return int|null
+     *
+     * @see    magento/module-customer/Model/Session.php
+     */
+    public function getMagentoCustomerId()
+    {
+        return $this->magentoCustomerSession->getCustomerId();
+    }
+
+    /**
+     * Create an Omise customer, then assigning its Omise customer id
+     * back to the customer's attribute who execute the method.
+     *
+     * @return self
+     */
+    public function create(array $params)
+    {
+        $this->customerAPI = $this->customerAPI->create($params);
+
+        $this->magentoCustomer->setData(self::OMISE_CUSTOMER_ID_FIELD, $this->customerAPI->id);
+        $this->magentoCustomerResource->saveAttribute($this->magentoCustomer, self::OMISE_CUSTOMER_ID_FIELD);
+
+        return $this;
+    }
+
+    /**
+     * Attach an Omise Token to a Customer object.
+     *
+     * @param string $cardToken
+     *
+     * @return self
+     */
+    public function addCard($cardToken)
+    {
+        if (! $this->getId()) {
+            $this->create(array(
+                'email'       => $this->magentoCustomer->getEmail(),
+                'description' => trim($this->magentoCustomer->getFirstname() . ' ' . $this->magentoCustomer->getLastname())
+            ));
+        }
+
+        $this->customerAPI = $this->customerAPI->update(array('card' => $cardToken));
+
+        return $this;
+    }
+
+    /**
+     * Retrieve all cards that belong to a specific Omise Customer object.
+     *
+     * @param  array $options
+     *
+     * @return \OmiseCardList
+     *
+     * @see    https://github.com/omise/omise-php/blob/master/lib/omise/OmiseCardList.php
+     */
+    public function cards($options = array())
+    {
+        return $this->customerAPI->cards($options);
+    }
+
+    /**
+     * @return array  of Omise card object.
+     *
+     * @see    https://www.omise.co/cards-api
+     */
+    public function getLatestCard()
+    {
+        $cards = $this->cards(array('order' => 'reverse_chronological'));
+
+        return $cards['total'] > 0 ? $cards['data'][0] : null;
+    }
+
+    /**
+     * Load and prepare Omise customer object
+     * when the class is being executed.
+     */
+    protected function initializeObject()
+    {
+        if ($this->getId()) {
+            $this->customerAPI = $this->customerAPI->find($this->getId());
+        }
+    }
+}
```

Comment:
Ordinarily, I would make `private` and `protected` member names start with an underscore, or failing that, follow the prevailing practice in a project. What is the general consensus in Magento 2? So far, I've seen a mixture. We should probably standardise? I would suggest using the underscore prefix

Reply:
Have noticed the same mess in our Magento 1 plugin. Fixing...

Reply:
As for Magento 2 standard n PSR-2, here we are choosing to go without `_` prefix. (let me know if you have spotted any place that using `_` prefix.)